app/controllers/api/databases.php

```diff
@@ -3339,25 +3321,23 @@ function updateAttribute(
                         $relation = new Document($relation);
                     }
                     if ($relation instanceof Document) {
-                        $oldDocument = Authorization::skip(fn() => $dbForProject->getDocument(
+                        $relatedDocumentOldVersion = Authorization::skip(fn() => $dbForProject->getDocument(
                             'database_' . $database->getInternalId() . '_collection_' . $relatedCollection->getInternalId(),
                             $relation->getId()
                         ));
 
-                        if ($oldDocument->isEmpty()) {
-                            $type = Database::PERMISSION_CREATE;
+                        // Attribute $collection is required for Utopia. Copying it from old version of document.
+                        $relation->setAttribute('$collection', $relatedDocumentOldVersion->getAttribute('$collection'));
 
+                        // If the child document has to be created it will need checking permissions.
+                        if ($relatedDocumentOldVersion->isEmpty()) {
                             if (isset($relation['$id']) && $relation['$id'] === 'unique()') {
                                 $relation['$id'] = ID::unique();
                             }
                         } else {
-                            $relation->removeAttribute('$collectionId');
-                            $relation->removeAttribute('$databaseId');
```

Comment:
We need to remove these attributes here, and add them back after updating, because Utopia doesn't know about them and will complain that the column doesn't exist